h2o-algos/src/main/java/hex/ensemble/Metalearner.java

```diff
@@ -25,366 +22,222 @@
 import water.fvec.Frame;
 import water.util.Log;
 
-class Metalearner {
```

Comment:
DRYing this class:
- providing factory method to retrieve metalearner instance.
- each algo may custom logic in its own impl class: GLM and AUTO are currently the main ones requiring custom logic.